server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -2000,17 +2000,13 @@ public Builder numberOfShards(int numberOfShards) {
          * Builder to create IndexMetadata that has an increased shard count (used for re-shard).
          * The new shard count must be a multiple of the original shardcount.
          * We do not support shrinking the shard count.
-         * @param shardCount   updated shardCount
-         *
-         * TODO: Check if this.version needs to be incremented
+         * @param targetShardCount   target shard count after resharding
          */
-        public Builder reshardAddShards(int shardCount) {
-            // Assert routingNumShards is null ?
-            // Assert numberOfShards > 0
-            if (shardCount % numberOfShards() != 0) {
+        public Builder reshardAddShards(int targetShardCount, final int sourceNumShards) {
```

Comment:
isn't `sourceNumShards` extractable from `this.settings`? Passing it in opens the door to the same kind of inconsistency I was worried about before when we were passing in the whole metadata.

Reply:
Thanks! Sorry about all the iteration.